backend/LexBoxApi/GraphQL/CustomTypes/ProjectGqlConfiguration.cs

```diff
@@ -10,7 +11,7 @@ protected override void Configure(IObjectTypeDescriptor<Project> descriptor)
         descriptor.Field(p => p.Code).IsProjected();
         descriptor.Field(p => p.CreatedDate).IsProjected();
         descriptor.Field(p => p.Id).Use<RefreshJwtProjectMembershipMiddleware>();
-        descriptor.Field(p => p.Users).Use<RefreshJwtProjectMembershipMiddleware>();
+        descriptor.Field(p => p.Users).Use<RefreshJwtProjectMembershipMiddleware>().Authorize(LexAuthPolicies.CanAccessProjectUsers);
```

Comment:
It's unclear to me from the [documentation](https://chillicream.com/docs/hotchocolate/v13/security/authorization) whether `.Authorize` runs as part of the middleware stack, or whether authorization is completely separate from the middleware run order. So the order of these calls may not matter; if it does, then it's just barely possible that a user who's just been assigned to a project might get a false negative here (denied access when he should have been granted it, because the project membership middleware has not yet refreshed the JWT before Authorize runs -- note that the middleware runs in opposite order of its declaration because it's calling `next(context) first thing), but that calling `.Authorize` before `.Use` would have avoided the false negative. I'm not sure how we'd devise a test for that scenario, though.

And it's probably moot since the `p.Id` reference is likely to come before the `p.Users` reference in the query, and that will cause a JWT refresh anyway.

Reply:
There's actually an enum you can pass to Authorize which you can use to specify when in the pipeline it will run. By default it will run before the resolver (aka middleware). It could also run after, but in most cases that would result in all the work being done to execute the query and then rejecting it, I'd rather not change that just for an edge case.